app/models/ability.rb

```diff
@@ -0,0 +1,38 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    #
+    #   return unless user.present?
+    #   can :read, :all
+    return unless user.present?
+
+    case user.role?
+    when 'Admin'
+      can :manage, :all
+    else
+      can :destroy, Post, author: user
+      can :destroy, Comment, author: user
```

Comment:
- [ ] please note that when the user is not an admin he can manage all of the posts and comments that he own , and not only deleting them kindly update your authorization rules 🙏 
example : 
```ruby
 when 'Admin'
      can :manage, :all
 else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
   end
```